core/src/commonMain/kotlin/State.kt

```diff
@@ -108,3 +110,34 @@ public sealed class State {
         }
     }
 }
+
+/**
+ * Returns `true` if `this` is at least in state [T], where [State]s are ordered:
+ * - [State.Disconnected] (smallest)
+ * - [State.Disconnecting]
+ * - [State.Connecting.Bluetooth]
+ * - [State.Connecting.Services]
+ * - [State.Connecting.Observes]
+ * - [State.Connected] (largest)
+ */
+internal inline fun <reified T: State> State.isAtLeast(): Boolean {
+    val currentState = when (this) {
+        is State.Disconnected -> 0
+        State.Disconnecting -> 1
+        State.Connecting.Bluetooth -> 2
+        State.Connecting.Services -> 3
+        State.Connecting.Observes -> 4
+        State.Connected -> 5
+        else -> error("Unreachable.")
+    }
+    val targetState = when (T::class) {
+        State.Disconnected::class -> 0
+        State.Disconnecting::class -> 1
+        State.Connecting.Bluetooth::class -> 2
+        State.Connecting.Services::class -> 3
+        State.Connecting.Observes::class -> 4
+        State.Connected::class -> 5
+        else -> error("Unreachable.")
```

Comment:
Same with here.

Reply:
I believe so, but it refused to compile (and this time it makes sense, because as far as Kotlin is concerned it can be _any_ `KClass`).